Vaadin application failed to start as Maven WAR build with dedicated Tomcat Server in the dev environment

### Description

Deploy our Vaadin application in development mod on tomcat server failed with the current exception since Vaadin 23.0.3:

```
java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/tms-we]]
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:916)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:843)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)

...

Caused by: java.lang.IllegalStateException: Failed to determine project directory for dev mode. Directory 'C:\Users\klehmann\Desktop' does not look like a Maven or Gradle project. Ensure that you have run the prepare-frontend Maven goal, which generates 'flow-build-info.json', prior to deploying your application
	at com.vaadin.base.devserver.startup.DevModeInitializer.getBaseDirectoryFallback(DevModeInitializer.java:383)
	at com.vaadin.base.devserver.startup.DevModeInitializer.initDevModeHandler(DevModeInitializer.java:211)
	at com.vaadin.base.devserver.DevModeHandlerManagerImpl.initDevModeHandler(DevModeHandlerManagerImpl.java:80)
```

Debugging the start process shows that with version 23.0.2 the project.basedir in DevModeInitialializer is set and with newer version it is null:

DevModeInitialializer line 208
`
String baseDir = config.getStringProperty(FrontendUtils.PROJECT_BASEDIR,
                null);
`



### Expected outcome

Vaadin application start in development mod without any exceptions

### Minimal reproducible example

No specific vaadin code required

### Steps to reproduce

1. Setup simple vaadin project, maven build, packagin war
2. Run project on dedicated apache tomcat server in development mod

### Environment

Vaadin version(s): Vaadin 23.0.3|4|5|6 (23.0.2 still working)
OS: Windows
IDE: Eclipse
Webserver: Tomcat 9.0.43 (managed in eclipse)
Springboot: no
Buildsystem: maven
Packaging: war

### Browsers

_No response_

--- a/flow-plugins/flow-maven-plugin/src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml
+++ b/flow-plugins/flow-maven-plugin/src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml
@@ -5,12 +5,24 @@
       <pluginExecutionFilter>
         <goals>
           <goal>validate</goal>
-          <goal>prepare-frontend</goal>
           <goal>build-frontend</goal>
         </goals>
       </pluginExecutionFilter>
       <action>
         <ignore></ignore>
+      </action>
+    </pluginExecution>
+    <pluginExecution>
+      <pluginExecutionFilter>
+        <goals>
+          <goal>prepare-frontend</goal>
+        </goals>
+      </pluginExecutionFilter>
+      <action>
+        <execute>
+          <runOnIncremental>false</runOnIncremental>
+          <runOnConfiguration>true</runOnConfiguration>
+        </execute>
       </action>
     </pluginExecution>
   </pluginExecutions>

